components/automate-gateway/gateway/datacollector.go

```diff
@@ -385,7 +385,7 @@ func getBoolIfExists(fieldName string, body []byte) bool {
 	fieldBytes, _, _, err := jsonparser.Get(body, fieldName)
 	if err == nil {
 		b, err := jsonparser.ParseBoolean(fieldBytes)
-		if err != nil {
+		if err == nil {
```

Comment:
This change fixes the bug.

Reply:
could you add a comment saying why we do the err == nil check here?  i know you included some info about it in the pr, but i think our future selves might appreciate having a comment in the code there